modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx

```diff
@@ -187,6 +188,67 @@ class StructNode extends React.Component {
     }
 
     /**
+     * Open JSON import dialog box and pass callback
+     */
+    onClickJsonImport() {
+        const onImport = (json) => {
+            let refExpr = TreeBuilder.build(FragmentUtils.parseFragment(FragmentUtils.createExpressionFragment(json)));
+            let currentValue;
+            this.props.model.setFields([], true);
+            refExpr.variable.initialExpression.keyValuePairs.forEach((ketValPair) => {
+                let currentName
+                if (TreeUtils.isLiteral(ketValPair.getKey())) {
+                    currentName = ketValPair.getKey().getValue().replace(/"/g, '');
+                } else {
+                    currentName = ketValPair.getKey().getVariableName().getValue();
+                }
+                if (TreeUtils.isRecordLiteralExpr(ketValPair.getValue())) {
+                    // TODO : Implement anonymous struct generation
+                    return;
+                } else {
+                    currentValue = ketValPair.getValue().getValue();
+                }
+
+                let currentType = 'string';
+                if (this.isInt(currentValue)) {
+                    currentType = 'int';
+                } else if (this.isFloat(currentValue)) {
+                    currentType = 'float';
+                }
+                refExpr = TreeBuilder.build(FragmentUtils.parseFragment(
+                  FragmentUtils.createStatementFragment(currentType + ' ' + currentName + ' = ' + currentValue + ';')));
+                this.props.model.addFields(refExpr.getVariable());
+            });
+        };
+        const id = 'composer.dialog.import.struct';
+        const { command: { dispatch } } = this.context;
+        dispatch('popup-dialog', {
+            id,
+            additionalProps: {
+                onImport,
+            },
+        });
+    }
+
+    /**
+     * Check given value is an Integer
+     * @param  {Number} val value to check
+     * @return {Boolean}  is Intger
+     */
+    isInt(val) {
```

Comment:
Can we avoid this individual type check? Or should it be for all primitive types? Not sure what we are doing here.

Reply:
Sample JSON may contain 23 or 23.5  to figure out if its a Integer or Float to generate struct types these checks are used.